Adding dbt Metrics Layer page (#998)
dbt-labs/docs.getdbt.com: website/docs/docs/dbt-cloud/cloud-metrics-layer.md

---
title: "What is the dbt Metrics Layer?"
id: "cloud-metrics-layer"
description: "The dbt metrics layer helps you standardize metrics within your organization."
keywords:
  - dbt metrics layer
---

> [dbt metrics](/docs/building-a-dbt-project/metrics) keeps your metrics consistent and provides a way to standardize them for an organization by abstracting metrics calculations out of the data warehouse and into the transformation later, which is where dbt exists. dbt metrics solves the problem of trying to calculate metrics along a set of specified dimensions accurately when metrics values are constantly changing.

Reviewer: this is a super-nit, but i think "metrics" here should be plural (like, the metrics themselves). "dbt metrics" is not going to be a singularized product name, we'll use that name to refer to the nodes in the dbt DAG themselves.

> by abstracting metrics calculations out of the data warehouse and into the transformation later, which is where dbt exists.

I really like this kind of framing! I think that "data warehouse" in this sentence kind of confuses the point though. Maybe we could position metrics in opposition to 1) pre-aggregation and 2) reimplementation in downstream data tools? So, something like:

> dbt metrics ensure metric consistency and provide a way to standardize metrics under version control in dbt projects. By abstracting metrics calculations out of pre-aggregated tables or specific BI tools, dbt metrics can be defined once and used everywhere. This helps ensure consistent reporting of key business metrics especially in an environment where metric definitions and dimensions are changing along with your business.

^ Feel free to workshop that - just a suggestion :)

Author: Addressed here: https://github.com/dbt-labs/docs.getdbt.com/pull/998/commits/79b74e0ef57dd36709435e4695677da326c9bfd2

> [dbt metrics](/docs/building-a-dbt-project/metrics) keeps your metrics consistent and provides a way to standardize them for an organization by abstracting metrics calculations out of the data warehouse and into the transformation later, which is where dbt exists. dbt metrics solves the problem of trying to calculate metrics along a set of specified dimensions accurately when metrics values are constantly changing.

Author: ```suggestion
[dbt metrics](/docs/building-a-dbt-project/metrics) ensure metric consistency and provide a way to standardize metrics under version control in dbt projects. By abstracting metrics calculations out of pre-aggregated tables or specific business intelligence tools (BI tools), dbt metrics can be defined once and used everywhere. This helps ensure consistent reporting of key business metrics especially in an environment where metric definitions and dimensions are changing along with your business.
```

> [dbt metrics](/docs/building-a-dbt-project/metrics) ensure metric consistency and provide a way to standardize metrics under version control in dbt projects. By abstracting metrics calculations out of pre-aggregated tables or specific business intelligence tools (BI tools), dbt metrics can be defined once and used everywhere. This helps ensure consistent reporting of key business metrics especially in an environment where metric definitions and dimensions are changing along with your business.

Author: ```suggestion
[dbt metrics](/docs/building-a-dbt-project/metrics) ensure metric consistency and provide a way to standardize metrics under version control in dbt projects. By abstracting metrics calculations out of pre-aggregated tables or specific business intelligence tools (BI tools), dbt metrics can be defined once and used everywhere. Defining metrics in one place ensures consistent reporting of key business metrics especially in an environment where metric definitions and dimensions are changing along with your business.
```

> Because dbt already transforms your data, it can use the same plane of existence to abstract out the metrics. dbt already knows about your core entities, for example, users, orders, and sessions. It can tell you not just about sessions, but average time on site. Or it might tell you not only about users but new users across these dimensions and this time range.

Reviewer: "plane of existence" is an amazing line haha

Author: You might not want to encourage me too much. Our docs will end up with the most [purple prose](https://en.wikipedia.org/wiki/Purple_prose) ever :)

Reviewer: purple prose for purple people if you ask me

> Because dbt already transforms your data, it can use the same plane of existence to abstract out the metrics. dbt already knows about your core entities, for example, users, orders, and sessions. It can tell you not just about sessions, but average time on site. Or it might tell you not only about users but new users across these dimensions and this time range.

Author: ```suggestion
dbt already transforms your data, so it can use the same plane of existence to abstract out the metrics. Because dbt already knows about your core entities, such as users, orders, and sessions, dbt metrics can reveal a broader story. You can learn not just about sessions, but about average time on site, or not only about users but about new users across these dimensions and this time range.
```

> [Sign up for the beta!](https://www.getdbt.com/signup/)

Reviewer: Could you link to https://forms.gle/4hi8YQ4mQ35QvYCh7 here? The CTA is to sign up for the beta, which isn't GA in dbt Cloud yet!

Author: Done! https://github.com/dbt-labs/docs.getdbt.com/pull/998/commits/392bd484e7a610b57e04ce7ab06690d9a72ad978

> [Sign up for the beta!](https://www.getdbt.com/signup/)

Author: ```suggestion
[Sign up for the beta!](https://forms.gle/4hi8YQ4mQ35QvYCh7)
```

> dbt already transforms your data, so it can use the same plane of existence to abstract out the metrics. Because dbt already knows about your core entities, such as users, orders, and sessions, dbt metrics can reveal a broader story. You can learn not just about sessions, but about average time on site, or not only about users but about new users across these dimensions and this time range.

Author: ```suggestion
dbt already transforms your data, so it can use the same plane of existence to abstract out the metrics. Because dbt already knows about your core entities, such as users, orders, and sessions, dbt metrics can reveal a broader story. You can learn not just about sessions, but about average time on site, or not only about users but about new users across xyz dimensions and in x time range.
```
